src/compiler/scala/tools/reflect/ReflectGlobal.scala

```diff
@@ -25,18 +25,14 @@ import scala.tools.nsc.typechecker.Analyzer
 class ReflectGlobal(currentSettings: Settings, reporter: Reporter, override val rootClassLoader: ClassLoader)
   extends Global(currentSettings, reporter) with scala.tools.reflect.ReflectSetup with scala.reflect.runtime.SymbolTable {
 
-  override lazy val analyzer = new {
-    val global: ReflectGlobal.this.type = ReflectGlobal.this
-  } with Analyzer {
-    /** Obtains the classLoader used for runtime macro expansion.
-     *
-     *  Macro expansion can use everything available in [[global.classPath]] or [[rootClassLoader]].
-     *  The [[rootClassLoader]] is used to obtain runtime defined macros.
-     */
-    override protected def findMacroClassLoader(): ClassLoader = {
-      val classpath = global.classPath.asURLs
-      ScalaClassLoader.fromURLs(classpath, rootClassLoader)
-    }
+  /** Obtains the classLoader used for runtime macro expansion.
+    *
+    *  Macro expansion can use everything available in `global.classPath` or `rootClassLoader`.
+    *  The `rootClassLoader` is used to obtain runtime defined macros.
+    */
+  override protected[scala] def findMacroClassLoader(): ClassLoader = {
```

Comment:
I'll see if I can move it back. My aim in moving it was to make it _easier_ to override in Zinc (just add to your subclass of `Global`, no need to have a subclass of `Analyzer` inside), but I see this is more pain than gain in 2.12.9.

Reply:
> Moving this outside of `analyzer` is a breaking change that would break the current zinc bridge that we use in bloop

@jvican meta-question: would the community build have caught that?

Reply:
I've moved this back to Analyzer in https://github.com/scala/scala/pull/7712/commits/7707a763fc09f38c760d10454696ca294fd1c0ec.

For 2.13.x, I plan to leave it in `Global`, but I have [removed](https://github.com/scala/scala/pull/7803) the `protected[scala]` qualifier so Zinc can override it.

Reply:
> meta-question: would the community build have caught that, if you hadn't caught it here?
  
Not yet because bloop's bridge is not in the community build, but I'd like in the future to make it a part of it. Usually this kind of changes is not a problem because we rarely override things from the compiler cake that change in minor compiler versions (changes in major compiler versions are fine).

> For 2.13.x, I plan to leave it in `Global`, but I have [removed](https://github.com/scala/scala/pull/7803) the `protected[scala]` qualifier so Zinc can override it.
  
Thanks a lot 🙏 It looks like what I wanted.